src/utils.cc

```diff
@@ -275,15 +277,20 @@ std::string minio::utils::Md5sumHash(std::string_view str) {
   return minio::utils::Base64Encode(hash);
 }
 
-std::string minio::utils::FormatTime(const std::tm* time, const char* format) {
+std::string minio::utils::FormatTime(const std::tm& time, const char* format) {
   char buf[128];
-  std::strftime(buf, 128, format, time);
+  std::strftime(buf, 128, format, &time);
   return std::string(buf);
 }
 
-std::tm* minio::utils::UtcTime::getBrokenDownTime() const {
-  const time_t secs = secs_;
-  return std::localtime(&secs);
+std::tm minio::utils::UtcTime::auxLocaltime(const std::time_t& time) {
+  std::tm result{};
+#ifdef _WIN32
+  localtime_s(&result, &time);
+#else
+  localtime_r(&time, &result);
```

Comment:
I see this function is available since C23 https://en.cppreference.com/w/c/chrono/localtime. As we set `set(CMAKE_CXX_STANDARD 17)` in CMakeLists.txt. Will it be a problem?

Reply:
Unfortunately, this is the C standard, not the C++ standard, so I'm afraid it won't help much. We'd need to check the actual C standard support in MSVC, clang and GCC to use these functions reliably because what the standard says is one thing and the actual coverage is another. The current implementation is "good enough" in the sense that it works on Ubuntu and Windows, but perhaps we should consider CMake-level discovery in a particular build environment.